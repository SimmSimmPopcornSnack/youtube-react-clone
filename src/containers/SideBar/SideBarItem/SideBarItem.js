import React from "react";
import {Icon, Menu} from "semantic-ui-react";
import { Link, withRouter } from "react-router-dom";
import "./SideBarItem.scss";

export class SideBarItem extends React.Component {
    render() {
        // React will ignore custom boolean attribute, therefore we pass a string
        // we use this attribute in our SCSS for styling
        const highlight = this.props.highlight ? 'highlight-item' : null;
        return (
            <Link to={{ pathname: this.props.path }}>
                <Menu.Item className={['sidebar-item', highlight].join(' ')}>
                    <div className="sidebar-item-alignment-container">
                        <span><Icon size="large" name={this.props.icon} /></span>
                        <span>{this.props.label}</span>
                    </div>
                </Menu.Item>
            </Link>
        );
    }
    shouldBeHighLighted() {
        const {pathname} = this.props.location;
        console.log(`pathname = `, pathname);
        if(this.props.path = "/") {
            return pathname === this.props.path;
        }
        return pathname.includes(this.props.path);
    }
}

export default withRouter(SideBarItem);
