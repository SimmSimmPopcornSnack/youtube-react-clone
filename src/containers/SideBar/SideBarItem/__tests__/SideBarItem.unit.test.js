import React from "react";
import { shallow } from "enzyme";
import { SideBarItem } from "../SideBarItem";

const location = {
    pathname: "/feed/trending",
};

describe("SideBarItem", () => {
    test("renders empty SideBarItem", () => {
        const wrapper = shallow(
            <SideBarItem location={location}/>
        );
        expect(wrapper).toMatchSnapshot();
    });
    
    test("renders highlighted SideBarItem that navigates to /feed/trending", () => {
        const wrapper = shallow(
            <SideBarItem highlight icon="fire" label="Trending" location={location} path={"/feed/trending"}/>
        );
        expect(wrapper).toMatchSnapshot();
    });
    
    test("renders non-highlighted SideBarItem that navigates to /feed/trending", () => {
        const wrapper = shallow(
            <SideBarItem icon="fire" label="Trending" location={location}/>
        );
        expect(wrapper).toMatchSnapshot();
    });
    
    test("renders highlighted SideBarItem with no navigaiton", () => {
        const wrapper = shallow(
            <SideBarItem highlight icon="fire" label="Trending" location={location}/>
        );
        expect(wrapper).toMatchSnapshot();
    });
});

