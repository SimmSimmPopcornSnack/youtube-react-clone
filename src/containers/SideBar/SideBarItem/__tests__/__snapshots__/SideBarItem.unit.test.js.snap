// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBarItem renders empty SideBarItem 1`] = `
<Link
  to={
    Object {
      "pathname": undefined,
    }
  }
>
  <MenuItem
    className="sidebar-item "
  >
    <div
      className="sidebar-item-alignment-container"
    >
      <span>
        <Icon
          as="i"
          size="large"
        />
      </span>
      <span />
    </div>
  </MenuItem>
</Link>
`;

exports[`SideBarItem renders highlighted SideBarItem that navigates to /feed/trending 1`] = `
<Link
  to={
    Object {
      "pathname": "/feed/trending",
    }
  }
>
  <MenuItem
    className="sidebar-item highlight-item"
  >
    <div
      className="sidebar-item-alignment-container"
    >
      <span>
        <Icon
          as="i"
          name="fire"
          size="large"
        />
      </span>
      <span>
        Trending
      </span>
    </div>
  </MenuItem>
</Link>
`;

exports[`SideBarItem renders highlighted SideBarItem with no navigaiton 1`] = `
<Link
  to={
    Object {
      "pathname": undefined,
    }
  }
>
  <MenuItem
    className="sidebar-item "
  >
    <div
      className="sidebar-item-alignment-container"
    >
      <span>
        <Icon
          as="i"
          name="fire"
          size="large"
        />
      </span>
      <span>
        Trending
      </span>
    </div>
  </MenuItem>
</Link>
`;

exports[`SideBarItem renders non-highlighted SideBarItem that navigates to /feed/trending 1`] = `
<Link
  to={
    Object {
      "pathname": undefined,
    }
  }
>
  <MenuItem
    className="sidebar-item "
  >
    <div
      className="sidebar-item-alignment-container"
    >
      <span>
        <Icon
          as="i"
          name="fire"
          size="large"
        />
      </span>
      <span>
        Trending
      </span>
    </div>
  </MenuItem>
</Link>
`;
